src/contacts/model/ContactModel.ts

```diff
@@ -178,6 +182,59 @@ export class ContactModel {
 				await this.loadContactLists()
 			}
 		}
+
+		if (isApp()) {
+			this.processNativeContactEntityEvents(updates)
+		}
+	}
+
+	private async processNativeContactEntityEvents(events: ReadonlyArray<EntityUpdateData>) {
+		const userId = this.loginController.getUserController().userId
+		const contactsIdToCreateOrUpdate: Map<Id, Array<Id>> = new Map()
+
+		for (const event of events) {
+			if (!isUpdateForTypeRef(ContactTypeRef, event)) continue
+			if (event.operation === OperationType.CREATE) {
+				if (contactsIdToCreateOrUpdate.has(event.instanceListId)) {
+					contactsIdToCreateOrUpdate.get(event.instanceListId)?.push(event.instanceId)
+				} else {
+					contactsIdToCreateOrUpdate.set(event.instanceListId, [event.instanceId])
+				}
+			} else if (event.operation === OperationType.UPDATE) {
+				if (contactsIdToCreateOrUpdate.has(event.instanceListId)) {
+					contactsIdToCreateOrUpdate.get(event.instanceListId)?.push(event.instanceId)
+				} else {
+					contactsIdToCreateOrUpdate.set(event.instanceListId, [event.instanceId])
+				}
+			} else if (event.operation === OperationType.DELETE) {
+				this.mobileSystemFacade?.deleteContact(userId, event.instanceId)
+			}
+		}
+
+		const contactsToInsertOrUpdate: StructuredContact[] = []
+
+		Promise.all(
+			Array.from(contactsIdToCreateOrUpdate.entries()).map(async (entries) => {
+				const contactList = await this.entityClient.loadMultiple(ContactTypeRef, entries[0], entries[1])
```

Comment:
I would do destructuring into `[listId, elementIds]`  in the argument itself, it's  more readable too I think